openflow15/openflow15.go

```diff
@@ -0,0 +1,2562 @@
+package openflow15
+
+// Package openflow15 provides OpenFlow 1.5 structs along with Read
+// and Write methods for each.
+// OpenFlow Wire Protocol 0x06
+//
+// Struct documentation is taken from the OpenFlow Switch
+// Specification Version 1.5
+// url https://opennetworking.org/wp-content/uploads/2014/10/openflow-switch-v1.5.1.pdf
+
+import (
+	"encoding/binary"
+	"errors"
+	"fmt"
+	"net"
+
+	"antrea.io/libOpenflow/common"
+	"antrea.io/libOpenflow/util"
+)
+
+const (
+	VERSION = 6
+)
+
+// Returns a new OpenFlow header with version field set to v1.5.
+var NewOfp15Header func() common.Header = common.NewHeaderGenerator(VERSION)
+
+// Echo request/reply messages can be sent from either the
+// switch or the controller, and must return an echo reply. They
+// can be used to indicate the latency, bandwidth, and/or
+// liveness of a controller-switch connection.
+func NewEchoRequest() *common.Header {
+	h := NewOfp15Header()
+	h.Type = Type_EchoRequest
+	return &h
+}
+
+// Echo request/reply messages can be sent from either the
+// switch or the controller, and must return an echo reply. They
+// can be used to indicate the latency, bandwidth, and/or
+// liveness of a controller-switch connection.
+func NewEchoReply() *common.Header {
+	h := NewOfp15Header()
+	h.Type = Type_EchoReply
+	return &h
+}
+
+// ofp_type 1.5
+const (
+	/* Immutable messages. */
+	Type_Hello        = 0
+	Type_Error        = 1
+	Type_EchoRequest  = 2
+	Type_EchoReply    = 3
+	Type_Experimenter = 4
+
+	/* Switch configuration messages. */
+	Type_FeaturesRequest  = 5
+	Type_FeaturesReply    = 6
+	Type_GetConfigRequest = 7
+	Type_GetConfigReply   = 8
+	Type_SetConfig        = 9
+
+	/* Asynchronous messages. */
+	Type_PacketIn    = 10
+	Type_FlowRemoved = 11
+	Type_PortStatus  = 12
+
+	/* Controller command messages. */
+	Type_PacketOut = 13
+	Type_FlowMod   = 14
+	Type_GroupMod  = 15
+	Type_PortMod   = 16
+	Type_TableMod  = 17
+
+	/* Multipart messages. */
+	Type_MultiPartRequest = 18
+	Type_MultiPartReply   = 19
+
+	/* Barrier messages. */
+	Type_BarrierRequest = 20
+	Type_BarrierReply   = 21
+
+	/* Controller role change request messages. */
+	Type_RoleRequest = 24
+	Type_RoleReply   = 25
+
+	/* Asynchronous message configuration. */
+	Type_GetAsyncRequest = 26
+	Type_GetAsyncReply   = 27
+	Type_SetAsync        = 28
+
+	/* Meters and rate limiters configuration messages. */
+	Type_MeterMod = 29
+
+	/* Controller role change event messages. */
+	Type_RoleStatus = 30
+
+	/* Asynchronous messages. */
+	Type_TableStatus = 31
+
+	/* Request forwarding by the switch. */
+	Type_RequestForward = 32
+
+	/* Bundle operations (multiple messages as a single operation). */
+	Type_BundleControl    = 33
+	Type_BundleAddMessage = 34
+
+	/* Controller Status async message. */
+	Type_ControllerStatus = 35
+)
+
+func Parse(b []byte) (message util.Message, err error) {
+	fmt.Println("Parse_Type=", b[1])
```

Comment:
Forgot to remove?

Reply:
Yeah. Had added during unit test. Would remove.